Cython/Shadow.py

```diff
@@ -164,13 +164,13 @@ def cmod(a, b):
 
 def cast(t, *args, **kwargs):
     kwargs.pop('typecheck', None)
-    assert not kwargs
 
     if isinstance(t, typedef):
-        return t(*args)
+        assert not kwargs
+        return t(*args, **kwargs)
```

Comment:
```suggestion
        assert not kwargs
        return t(*args)
```

Reply:
I have not yet applied these changes, see the comment above. It should be fixed by doing this (which is both of your suggestions):
```diff
diff --git a/Cython/Shadow.py b/Cython/Shadow.py
index 6e0bbcd07..b73bfee6e 100644
--- a/Cython/Shadow.py
+++ b/Cython/Shadow.py
@@ -164,13 +164,13 @@ def cmod(a, b):

 def cast(t, *args, **kwargs):
     kwargs.pop('typecheck', None)
+    assert not kwargs

     if isinstance(t, typedef):
-        assert not kwargs
-        return t(*args, **kwargs)
+        return t(*args)
     elif isinstance(t, type):  # Doesn't work with old-style classes of Python 2.x
         if len(args) != 1 or not (args[0] is None or isinstance(args[0], t)):
-            return t(*args, **kwargs)
+            return t(*args)

     return args[0]
```